### PR TITLE
[DOCS] Reformat compound word token filters

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the compound token filters docs as part of #44726.

I also decided to separate the token filter docs as part of this PR. However, I don't feel strongly about it. If we prefer to keep these token filter docs on the same page, I can do that.

### Changes
* Separates the compound token filters doc pages into separate token
  filter pages:
  * Dictionary decompounder token filter
  * Hyphenation decompounder token filter

* Adds analyze API examples for each compound token filter

* Adds a redirect for the removed compound token filters page

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+included in the token output.
+
+Either this or the `word_list_path` parameter is required.
+--
+
+`word_list_path`::
++
+--
+(Required+++*+++, string)
+Path to a file containing a list of subwords to find in the token stream. If
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc]:
```suggestion
Path to a file that contains a list of subwords to find in the token stream. If
```
Getting into picky grammar-tics that are more prescriptive than truly necessary.

File: docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+found, the subword is included in the token output.
+
+This path must be absolute or relative to the `config` location. The file must
+be UTF-8 encoded. Each token in the file must be separated by a line break.
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc]: I'd be inclined to combine the first two sentences:

>This path must be absolute or relative to the `config` location, and the file must be UTF-8 encoded.

File: docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+Either this or the `word_list` parameter is required.
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc]:
```suggestion
Either this parameter or `word_list` must be specified.
```

File: docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc [hunk 4]
```diff
@@ -0,0 +1,154 @@
+[[analysis-hyp-decomp-tokenfilter]]
+=== Hyphenation decompounder token filter
+++++
+<titleabbrev>Hyphenation decompounder</titleabbrev>
+++++
+
+Uses XML-based hyphenation patterns to find potential subwords in compound
+words. These subwords are then checked against a provided word list. Subwords not
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc]:
```suggestion
words. These subwords are then checked against the specified word list. Subwords not
```

File: docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
++
+--
+(Required+++*+++, array of strings)
+A list of subwords. Subwords found using the hyphenation pattern but not in this
+list are excluded from the token output.
+
+You can use the <<analysis-dict-decomp-tokenfilter,`dictionary_decompounder`>>
+filter to test the quality of word lists before implementing them.
+
+Either this or the `word_list_path` parameter is required.
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc]:
```suggestion
Either this parameter or `word_list_path` must be specified.
```

File: docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+`word_list_path`::
++
+--
+(Required+++*+++, string)
+Path to a file containing a list of subwords. Subwords found using the
+hyphenation pattern but not in this list are excluded from the token output.
+
+This path must be absolute or relative to the `config` location. The file must
+be UTF-8 encoded. Each token in the file must be separated by a line break.
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc]: Same as above.

File: docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+You can use the <<analysis-dict-decomp-tokenfilter,`dictionary_decompounder`>>
+filter to test the quality of word lists before implementing them.
+
+Either this or the `word_list` parameter is required.
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc]:
```suggestion
Either this parameter or `word_list` must be specified.
```

File: docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+basis for a new custom token filter. You can modify the filter using its
+configurable parameters.
+
+For example, the following <<indices-create-index,create index API>> request
+uses a custom `hyphenation_decompounder` filter to configure a new
+<<analysis-custom-analyzer,custom analyzer>>.
+
+The custom `hyphenation_decompounder` filter find subwords based on hyphenation
+patterns in the `analysis/hyphenation_patterns.xml` file. The filter then checks
+these subwords against a provided list of words in the
```
Comment (debadair) [inline docs/reference/analysis/tokenfilters/hyphenation-decompounder-tokenfilter.asciidoc]:
```suggestion
these subwords against the list of words specified in the
```

**Review by debadair — APPROVED** [review]
Left some minor suggestions. I think for reference items, splitting them apart is better for discoverability.

**Review by romseygeek — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
@elasticsearch update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**